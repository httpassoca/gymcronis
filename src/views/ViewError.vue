<template>
<div class="error">
  <Card>
    <div v-if="!error">
      <h2>{{`Page "${$router.currentRoute.path}" does not exist`}}</h2>
      <p>Bro, here is lonely and empty. Go to a better place ü§ú</p>
      <router-link :to="{name: 'Workouts'}">
        <Button type="primary"> Go to workouts page </Button>
      </router-link>
    </div>
    <div v-else-if="error === '401'">
      <h2> Unauthorized üîê </h2>
      <p>Seems you dont have access to this page...</p>
      <router-link :to="{name: 'Auth'}">
        <Button type="primary"> Go to login page </Button>
      </router-link>
    </div>
  </Card>
</div>
</template>

<script lang="ts">
import Vue from 'vue';
import { Button, Card } from 'element-ui';

export default Vue.extend({
  name: 'ViewError',

  components: {
    Button,
    Card,
  },

  props: {
    error: {
      type: String,
    },
  },
});
</script>

<style lang="sass" scoped>
@import '../sass/mixins'

.error
  @include visitorPage
  p
    margin: 12px 0
</style>
