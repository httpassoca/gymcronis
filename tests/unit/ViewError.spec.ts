import { mount, createLocalVue } from '@vue/test-utils';
import ElementUI from 'element-ui';
import ViewError from '../../src/views/ViewError.vue';

const localVue = createLocalVue();
localVue.use(ElementUI);

describe('ViewError.vue', () => {
  const $router = {
    currentRoute: { path: 'wrongPage' },
  };

  const factory = (options?: any) => (
    mount(ViewError, {
      localVue,
      ...options,
      mocks: { $router },
      stubs: ['router-link'],
    })
  );

  test('should render component', () => {
    expect(!!factory().html()).toBe(true);
  });

  test('should show the wrong page when 404 error', () => {
    const wrapper = factory();
    expect(wrapper.get('h2').text()).toContain('Page "wrongPage" does not exist');
  });

  test('should show "Unauthorized" when 401 error', () => {
    const wrapper = factory({ propsData: { error: '401' } });
    expect(wrapper.get('h2').text()).toContain('Unauthorized');
  });
});
