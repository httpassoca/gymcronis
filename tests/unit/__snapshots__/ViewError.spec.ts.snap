// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewError.vue should render component 1`] = `
<div class="error">
  <div class="el-card is-always-shadow">
    <!---->
    <div class="el-card__body">
      <div>
        <h2>Page "wrongPage" does not exist</h2>
        <p>Bro, here is lonely and empty. Go to a better place ðŸ¤œ</p>
        <router-link-stub to="[object Object]"><button type="button" class="el-button el-button--primary">
            <!---->
            <!----><span> Go to workouts page </span>
          </button></router-link-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewError.vue should show "Unauthorized" when 401 error 1`] = `"received is not a function"`;
